ComSemApp/templates/ComSemApp/student/assessment.html

```diff
@@ -3,20 +3,52 @@
 {% block content %}
 <h1>Speaking Practice View</h1>
 
+<p><span id="sentence_text" ></span></p>
+<p id="text_for_timer"><span id="text_timer"> </span> </p>
+<progress id="timer_bar"></progress>
+
 <script>
-    var given_time = 15;
-    var current_time =15;
-    var question_timer = setInterval(function(){
-      if(current_time <= 0){
-        clearInterval(question_timer);
+
+const reading_offset = 2;
+const recording_offset = 3;
+let sentence = 'This an example.'; // Change this to desired expression
+document.getElementById("sentence_text").textContent = sentence
+document.getElementById("timer_bar").style.visibility = "hidden";
+document.getElementById("text_for_timer").style.visibility = "hidden";
+
+P = timer(sentence, reading_offset);//runs the reading timer
+P.then(() => {//Once reading timer done this starts
+  //This is where recording start goes
+  P2= timer(sentence, recording_offset);//Runs the recording timer
+  P2.then(() => { //Once recording timer done this starts
+    //This is recording stop goes
+    document.getElementById("sentence_text").textContent = 'Problem Complete!';//This was a last test can be deleted
+  });
+});
+
+function timer(sentence, base_time){
+  return new Promise(function(resolve, reject){
```

Comment:
The anonymous function for the promise should be declared "(resolve, reject) => {". The function keyword works but, as far as I know, is generally frowned upon.